From: Marvin Scholz <epirat07@gmail.com>
Date: Thu, 5 Feb 2026 16:34:35 +0100
Subject: modules: spatialaudio: compat with new API

This merely makes it build with the new API and doesn't yet migrate
to newer APIs.

(cherry picked from commit a41efcaf1d0b799f0d85d2ca0c59a9c677153cc4) (rebased)
rebased:
- VLC 3 still has the spatialaudio-headphones option
Signed-off-by: Steve Lhomme <robux4@ycbcr.xyz>
---
 .../audio_filter/channel_mixer/spatialaudio.cpp    | 39 +++++++++++++++++++---
 1 file changed, 35 insertions(+), 4 deletions(-)

diff --git a/modules/audio_filter/channel_mixer/spatialaudio.cpp b/modules/audio_filter/channel_mixer/spatialaudio.cpp
index ec3b0e7..311ab0c 100644
--- a/modules/audio_filter/channel_mixer/spatialaudio.cpp
+++ b/modules/audio_filter/channel_mixer/spatialaudio.cpp
@@ -43,6 +43,23 @@
 #include <spatialaudio/Ambisonics.h>
 #include <spatialaudio/SpeakersBinauralizer.h>
 
+#ifdef HAVE_SPATIALAUDIOVERSION_H
+#  include <spatialaudio/SpatialaudioVersion.h>
+#else
+#  define SPATIALAUDIO_API_VERSION_MAJOR 0
+#endif
+
+#if SPATIALAUDIO_API_VERSION_MAJOR >= 2
+    using namespace spaudio;
+    using CAmbisonicBinauralizer = spaudio::AmbisonicBinauralizer;
+    using CAmbisonicDecoder = spaudio::AmbisonicDecoder;
+    using CAmbisonicProcessor = spaudio::AmbisonicProcessor;
+    using CAmbisonicZoomer = spaudio::AmbisonicZoomer;
+    using CAmbisonicSpeaker = spaudio::AmbisonicSpeaker;
+    using CBFormat = spaudio::BFormat;
+    #define kAmblib_CustomSpeakerSetUp Amblib_SpeakerSetUps::kAmblib_CustomSpeakerSetUp
+#endif
+
 #define CFG_PREFIX "spatialaudio-"
 
 #define DEFAULT_HRTF_PATH "hrtfs" DIR_SEP "dodeca_and_7channel_3DSL_HRTF.sofa"
@@ -460,10 +477,18 @@ static int Open(vlc_object_t *p_this)
         p_sys->mode = filter_spatialaudio::AMBISONICS_DECODER;
 
         unsigned i_nbChannels = aout_FormatNbChannels(&p_filter->fmt_out.audio);
-        if (i_nbChannels == 1
-         || !p_sys->speakerDecoder.Configure(p_sys->i_order, true,
-                                             kAmblib_CustomSpeakerSetUp,
-                                             i_nbChannels))
+        bool ok = false;
+        if (i_nbChannels > 1) {
+#if SPATIALAUDIO_API_VERSION_MAJOR >= 2
+            ok = p_sys->speakerDecoder.Configure(p_sys->i_order, true,
+                AMB_BLOCK_TIME_LEN, p_filter->fmt_in.audio.i_rate,
+                kAmblib_CustomSpeakerSetUp, i_nbChannels);
+#else
+            ok = p_sys->speakerDecoder.Configure(p_sys->i_order, true,
+                kAmblib_CustomSpeakerSetUp, i_nbChannels);
+#endif
+        }
+        if (!ok)
         {
             msg_Err(p_filter, "Error creating the Ambisonics decoder.");
             delete p_sys;
@@ -509,7 +534,13 @@ static int Open(vlc_object_t *p_this)
         return VLC_EGENERIC;
     }
 
+#if SPATIALAUDIO_API_VERSION_MAJOR >= 2
+    if (!p_sys->zoomer.Configure(p_sys->i_order, true,
+                                 AMB_BLOCK_TIME_LEN,
+                                 p_filter->fmt_in.audio.i_rate))
+#else
     if (!p_sys->zoomer.Configure(p_sys->i_order, true, 0))
+#endif
     {
         msg_Err(p_filter, "Error creating the ambisonic zoomer.");
         delete p_sys;
